### PR TITLE
Openwrt base image v2

### PR DESCRIPTION
**Pull request by troth**
With some help from Aaron Brice (ambrice), we have gotten Luci to build and can log into the web gui running Luci. There's still more to do to get a functioning OpenWRT system from meta-openwrt, but this is progress.

This PR also includes a fix for issue #4. Aaron came up with a better fix than the one that I provided in the discussion in the issue.

Both Aaron and I have tested this and have been able to log into the Luci UI via a browser.

One issue I hit was the default qemux86 kernel didn't have CONFIG_BRIDGE=y configured. This caused netifd to fail to bring up the br-lan bridge interface due to kmodloader failing to load the bridge.ko module at boot time. I have not included my kernel bbappend changes in this patch series since I don't know the best approach for this. Basically I added the following to meta-openwrt:

```
$ git diff --cached
diff --git a/recipes-kernel/linux/linux-yocto/bridge.cfg b/recipes-kernel/linux/linux-yocto/bridge.cfg
new file mode 100644
index 0000000..873cbf3
--- /dev/null
+++ b/recipes-kernel/linux/linux-yocto/bridge.cfg
@@ -0,0 +1 @@
+CONFIG_BRIDGE=y
diff --git a/recipes-kernel/linux/linux-yocto_4.8.bbappend b/recipes-kernel/linux/linux-yocto_4.8.bbappend
new file mode 100644
index 0000000..18ac24c
--- /dev/null
+++ b/recipes-kernel/linux/linux-yocto_4.8.bbappend
@@ -0,0 +1,3 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
+
+SRC_URI += "file://bridge.cfg"
```

I suspect that should really be version agnostic... that is for another patch and another discussion.

### CODE REVIEW

File: recipes-core/netifd/netifd/network.config [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright (C) 2006 OpenWrt.org
```
Comment (kraj) [inline recipes-core/netifd/netifd/network.config]: whats the license for this file ?

File: recipes-core/netifd/netifd_git.bb [hunk 2]
```diff
@@ -8,9 +8,13 @@ LIC_FILES_CHKSUM = "file://main.c;beginline=1;endline=13;md5=572cd47ba0e377b2633
 SECTION = "base"
 DEPENDS = "json-c libubox ubus libnl uci"
 
-SRCREV = "64a655d8ffa9f0cea1bbdd35cac6b3b99b865270"
+SRCREV_netifd = "64a655d8ffa9f0cea1bbdd35cac6b3b99b865270"
+SRCREV_openwrt = "v15.05.1"
```
Comment (kraj) [inline recipes-core/netifd/netifd_git.bb]: its better to use raw SHA instead of tag here

File: recipes-core/netifd/netifd_git.bb [hunk 3]
```diff
@@ -19,3 +23,28 @@ inherit cmake pkgconfig openwrt
...
 OECMAKE_C_FLAGS += "-I${STAGING_INCDIR}/libnl3 -Wno-error=cpp"
+
+FILES_${PN} += "\
+    /usr/share/udhcpc/default.script \
+    /lib/netifd/dhcp.script \
+    /lib/netifd/utils.sh \
+    /lib/netifd/netifd-wireless.sh \
+    /lib/netifd/netifd-proto.sh \
+    /lib/netifd/proto/dhcp.sh \
+    /lib/network/config.sh \
```
Comment (kraj) [inline recipes-core/netifd/netifd_git.bb]: generally its better to use bitbake variables whereever possible.

**Comment by mattsm** [comment]
You guys consider setting up a mailing list for this pursuit? I'd be interested in tracking progress and helping out some.

**Comment by kraj** [comment]
We can use yocto ml with [meta-openwrt] subject prefix.

**Pull request merged**